plugin.json

```diff
@@ -98,6 +98,12 @@
                 "type": "text",
                 "help_text": "This should look something like \"googlebd09b1075898e210.html\"",
                 "default": ""
+            },
+            {
+                "key": "EncryptionKey",
+                "display_name": "Encryption key",
+                "type": "generated",
+                "help_text": "**Required**: The encryption key used to store data in the database"
```

Comment:
We'll want to think about backwards compatibility here for the mscalendar plugin. When we introduce this in a plugin, we typically release a new major version, which then links the admin to release notes before upgrading the plugin, so they can see what the breaking change is.

Unfortunately this will require all of their users to reconnect. Regenerating this value results in the same situation of data that cannot be unencrypted. One solution we've used is to DM all connected users to reconnect their accounts.

Reply:
> We could also allow a sysadmin command to load, encrypt and store the data again, to alleviate the pain a bit.

If we can make it turnkey for the admin, then I'm up for this. Note that some cloud customers are using mscalendar, so having the admin do anything database-level is not possible there.

We could also just do a migration on plugin start up. Here's an example of a migration using the kv store and `cluster.Mutex` https://github.com/mattermost/mattermost-plugin-gitlab/pull/361. Note that the migration would be a bit different for the mscalendar plugin, in that we would just be encrypting the data, and not sending user DMs etc. that the linked migration is doing. We would also want to make it so an encryption secret is automatically assigned if it is blank on plugin start up.